packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolAncestors.test.ts

```diff
@@ -0,0 +1,73 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import BigNumber from 'bignumber.js'
+import { MempoolTx } from '../../../src/category/blockchain'
+import waitForExpect from 'wait-for-expect'
+import { Testing } from '@defichain/jellyfish-testing'
+
+describe('getMempoolAncestors', () => {
+  const container = new MasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should return no ancestor ids for tx without ancestors', async () => {
+    const txId = await testing.rpc.wallet.sendToAddress('mwsZw8nF7pKxWH8eoKL9tPxTpaFkz7QeLU', 0.0003)
+    const mempoolEntry: string[] = await testing.rpc.blockchain.getMempoolAncestors(txId, false)
+    await waitForExpect(async () => {
+      expect(mempoolEntry.length).toStrictEqual(0)
+    }, 10000)
+  })
+
+  it('should return JSON object if verbose is true', async () => {
+    for (let i = 0; i < 3; i++) {
+      await testing.rpc.wallet.sendToAddress('mwsZw8nF7pKxWH8eoKL9tPxTpaFkz7QeLU', 0.0003)
+    }
+    const txId = await testing.rpc.wallet.sendToAddress('mwsZw8nF7pKxWH8eoKL9tPxTpaFkz7QeLU', 0.0003)
+    const mempoolEntry: MempoolTx = await testing.rpc.blockchain.getMempoolAncestors(txId, true)
+    const entryKey = Object.keys(mempoolEntry)[0]
+    await waitForExpect(async () => {
+      expect(Object.keys(mempoolEntry).length).toBeGreaterThan(0)
```

Comment:
```suggestion
    await waitForExpect(async () => {
        expect(Object.keys(mempoolEntry).length).toBeGreaterThan(0)
        const entryKey = Object.keys(mempoolEntry)[0]
```


We should expect that there is a single key before trying to access, if not the test will fail for unexpected reasons

Reply:
Ah, good catch. Thanks!